### PR TITLE
ARGO-1301 Refactor service-type - Add an additional field named type

### PR DESCRIPTION
**Pull request by agelostsal**

### CODE REVIEW

**Comment by kkoumantaros** [comment]
Pls update title to something meaningful aligned with Jira tickets

**Comment by kkoumantaros** [comment]
Pls update title to something meaningful aligned with Jira tickets

**Comment by kkoumantaros** [comment]
I'm not sure I understand why we need this step

**Comment by kkoumantaros** [comment]
Please add a better title for this PR

File: config/configuration-test-files/test-conf.json [hunk 1]
```diff
@@ -8,5 +8,6 @@
   "service_token": "token",
   "supported_auth_types": ["x509", "oidc"],
   "supported_auth_methods": ["api-key", "x-api-token"],
+  "supported_service_types": ["ams", "web-api", "custom"],
```
Comment (themiszamani) [inline config/configuration-test-files/test-conf.json]: Here the supported service types are ["ams", "web-api", "custom"],
but in readme and swagger you refer to 
+   "supported_service_types": ["ams", "web-api"],

Comment (agelostsal) [reply config/configuration-test-files/test-conf.json]: updated readme and swagger

**Pull request merged**